### DISCUSSION TITLE
v0.6.0 - Latent Consistency Model Support

### DISCUSSION THREAD

**Discussion by saddam213** [saddam213/OnnxStack#7] [General]
# Latent Consistency Models (LCM)
### LCM ImageToImage and TextToImage now supported
Support has been added for Latent Consistency Models, these model dramatically reduce the amount of steps required to generate an image.

Plus a few more changes in this weeks build
1. Multiple pipeline support
2. Latent Consistency Model support
3. Improved BetaSchedule calculation
4. Support disabled classifier free guidance
5. Batch Prompt, multiple images per prompt now supported
6. DDIM, EulerDiscrete, KDPM2 and LCM schedulers added

Big thanks to @luosiallen for this awesome model, below is a link to the LCM project
https://github.com/luosiallen/latent-consistency-model

## New Contributors
* @TheyCallMeHex made their first contribution in https://github.com/saddam213/OnnxStack/pull/5

**Full Changelog**: https://github.com/saddam213/OnnxStack/compare/v0.5.0...v0.6.0

## Nuget Packages
https://www.nuget.org/packages/OnnxStack.Core/0.6.0
https://www.nuget.org/packages/OnnxStack.StableDiffusion/0.6.0

<hr /><em>This discussion was created from the release <a href='https://github.com/saddam213/OnnxStack/releases/tag/v0.6.0'>v0.6.0 - Latent Consistency Model Support</a>.</em>

**Comment by JohnClaw** [comment]
Could you add this upscaler (https://huggingface.co/ssube/stable-diffusion-x4-upscaler-onnx), please?

**Reply by saddam213** [reply]
Hi John, upscaler is next on my list as I wait for my new GPU's to be delivered so I can continue with AnimatedDiff

Hopefully it should be added this weekend, if I dont get it done in time for Fridays release it should be ready for the following weeks release :)

**Reply by Amin456789** [reply]
hey Adam, thats great! so excited for animatediff! 
im using OnnxStack from now on only!
i have some suggestions and it will be great if we can have them someday:

1.could u please add a dark mode too? 

2. a more resizable ui window = i am using a laptop so i cant see generate button and i have to use an app named altdrag to force it and bring the app more upper as i cant see the generate fully, i can see a bit of it without altdrag but im afraid in the future i will miss new features if something goes down there and i cant see it

3.batch count for generating images one after one [ u mentioned batch prompt but i cant find it, im using sd 1.5]

4.saving all images in an output folder so we don't need to save one by one

thanks

**Reply by saddam213** [reply]
1. Dark theme will be included at some point, but its not a priority right now, once I have finalized the UI it wont take long to make one
2. I will get this done for Fridays release, what is you screen resolution? I will use that to make sure all controls can be reached :)
3. Batch support was added to the backend but I have not added support to any of the UI's yet, will try get it in this week
4. I will add a "Save All" button and also a checkbox "Auto Save" and a configurable "Save Directory"

The UI is still very new and very basic, it will get better as it progresses, but I am focused more on adding backend functionality right now, but will do my best to add requests :)

**Reply by Amin456789** [reply]
1. that will be great! :D i love dark them
2.  my resolution is 1366x768
3.great! cant wait for it!
4.great idea! this and batch count will get along so good, if i say correctly the pictures are being save in the ram for now right? ur new update will be so good to free up ram time to time and great for batch count
over all i love onnxstack, thank to u and @TheyCallMeHex for working on this! it runs so smooth and the ui looks great, im addicted to it now :D

**Reply by saddam213** [reply]
This Fridays release will simplify model loading, and support model downloads via http and git clone, also one click load from local folder

![image](https://github.com/saddam213/OnnxStack/assets/4353684/9dbf1cd5-4d01-45fd-8259-0452fbd3fad1)

@TheyCallMeHex is busy testing it now :)

**Reply by Amin456789** [reply]
i love it!

**Reply by saddam213** [reply]
Got 1,2 and 4 done, ran out of time to get batch and upscaler in, but 99% sure they will be done for next weeks release

v0.7.0 will be ready for download tomorrow :)

**Reply by saddam213** [reply]
![image](https://github.com/saddam213/OnnxStack/assets/4353684/54de2def-9849-4847-a600-933db196a571)

Dark theme is a million time nicer on the eyes

**Reply by Amin456789** [reply]
thank u! yes it looks amazing! cant wait for it!

**Comment by Amin456789** [comment]
hey guys im back, great job Adam!
@TheyCallMeHex can u please make some fp16 models too? lyriel v1.6 and deliberate v2 or 3 will be great!
https://huggingface.co/stablediffusionapi/deliberate-v3/tree/main
https://huggingface.co/pagebrain/lyriel-v1-6
thanks

**Reply by TheyCallMeHex** [reply]
What's the lyriel one? It just has an empty readme?

https://huggingface.co/TheyCallMeHex/Deliberate-V3-ONNX

**Reply by Amin456789** [reply]
thanks buddy!
here a civitai one=
https://civitai.com/models/22922/lyriel
or
https://huggingface.co/nyxia/lyriel16/tree/main

**Comment by Amin456789** [comment]
btw it will be great if we can have fp16 models [they are half the size and use lower vram/ram and speed up things i guess]
i tried to convert deliberate onnx to fp16 with this guide but failed :D
https://onnxruntime.ai/docs/performance/model-optimizations/float16.html
got errors like = ImportError: cannot import name 'onnx_pb'  at first but fixed it
then when it was to about convert unet model it gave me this =  Message onnx.ModelProto exceeds maximum protobuf size of 2GB: 3439988952
after sometime i found out even onnx says they wont go over 2gb for quantization... but it seems optimum can do it or script of diffusers named convert_stable_diffusion_checkpoint_to_onnx.py we have to add --fp16 [but it wants cuda gpu i think], but it can be done with this too = https://github.com/Amblyopius/Stable-Diffusion-ONNX-FP16

@TheyCallMeHex  can u please convert some of the models [lyriel and deliberate will be great] to fp16 and share if u r interested? it is getting too technical for me as i get errors left and right

Update: onnxstack works with quantizated int8 models too! even though quality is lower but it will be great if i can have some fp16 models to test

**Comment by Amin456789** [comment]
another LCM model released, didnt test it myself, i think it is anime related:
https://huggingface.co/furusu/LCM-Acertainty
